src/useSpring.ts

```diff
@@ -1,31 +1,37 @@
-import { useEffect, useState } from 'react';
+import { useEffect, useState, useMemo } from 'react';
 import { Spring, SpringSystem } from 'rebound';
 
 const useSpring = (targetValue: number = 0, tension: number = 50, friction: number = 3) => {
   const [spring, setSpring] = useState<Spring | null>(null);
   const [value, setValue] = useState<number>(targetValue);
 
-  useEffect(() => {
-    const listener = {
+  // memoize listener to being able to unsubscribe later properly, otherwise
+  // listener fn will be different on each re-render and wouldn't unsubscribe properly.
+  const listener = useMemo(
```

Comment:
I found couple of issues on this one by adding tests! First one: listener has to be defined out of useEffect and memoized, not for performance but for keeping same function within listener const so it can be referenced properly when unsubscribing (otherwise, as explained on the code comment, listener fn never would be the same as it generates a different one on each update).